src/scheduler/fifo.c

```diff
@@ -2450,6 +2450,16 @@ next_job(status *policy, server_info *sinfo, int flag)
 static void
 init_sc_attrs(void)
 {
+	static int first_time = 1;
+
+	if (!first_time) {
+		free(sc_attrs.comment);
+		free(sc_attrs.job_sort_formula);
+		free(sc_attrs.partition);
+		free(sc_attrs.sched_log);
+		free(sc_attrs.sched_port);
+		free(sc_attrs.sched_priv);
+	}
```

Comment:
If the above elements default to NULL, you don't need first_time.  Doing a free(NULL) is as noop.

Reply:
No, this is the only place where I initialize them, I figured that I had to do it every-time so I put it in a function instead of a file.

Reply:
It looks like sc_attrs is a global structure (not mallo()'d).  This means it is zeroed when it is declared.  You should be safe just freeing these.

Reply:
ok, i was a bit unsure if that was reliable or not. I'll remove it then, thanks.